internal/services/iotoperations/iotoperations_broker_listener_resource.go

```diff
@@ -0,0 +1,716 @@
+package iotoperations
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/iotoperations/2024-11-01/brokerlistener"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type BrokerListenerResource struct{}
+
+var _ sdk.ResourceWithUpdate = BrokerListenerResource{}
+
+type BrokerListenerModel struct {
+	Name                 string                    `tfschema:"name"`
+	ResourceGroupName    string                    `tfschema:"resource_group_name"`
+	InstanceName         string                    `tfschema:"instance_name"`
+	BrokerName           string                    `tfschema:"broker_name"`
+	ExtendedLocationName *string                   `tfschema:"extended_location_name"`
+	ExtendedLocationType *string                   `tfschema:"extended_location_type"`
+	ServiceName          *string                   `tfschema:"service_name"`
+	ServiceType          *string                   `tfschema:"service_type"`
+	Ports                []BrokerListenerPortModel `tfschema:"ports"`
+	ProvisioningState    *string                   `tfschema:"provisioning_state"`
```

Comment:
The resource model does not use pointer types; it exclusively utilizes literal types. this also applies to all other resource model definitions.